hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/security/alias/TestCredentialProviderFactory.java

```diff
@@ -250,18 +247,14 @@ public void testLocalJksProvider() throws Exception {
   }
 
   @Test
-  public void testLocalBCFKSProvider() throws Exception {
+  public void testLocalBCFKSProvider() {
     Configuration conf = new Configuration();
     final Path ksPath = new Path(tmpDir.toString(), "test.bcfks");
     final String ourUrl = LocalBouncyCastleFipsKeyStoreProvider.SCHEME_NAME +
         "://file" + ksPath.toUri();
     conf.set(CredentialProviderFactory.CREDENTIAL_PROVIDER_PATH, ourUrl);
-
-    exception.expect(IOException.class);
-    exception.expectMessage("Can't create keystore");
-    List<CredentialProvider> providers =
-        CredentialProviderFactory.getProviders(conf);
-    assertTrue("BCFKS needs additional JDK setup", providers.isEmpty());
+    assertThrows("Can't create keystore", IOException.class,
+        () -> CredentialProviderFactory.getProviders(conf));
```

Comment:
It seems not to validate the error message. Per the [junit docs](https://junit.org/junit5/docs/current/user-guide/#writing-tests-assertions),
```suggestion
    Exception exception = assertThrows(IOException.class,
        () -> CredentialProviderFactory.getProviders(conf));
    assertEquals("Can't create keystore", exception.getMessage());
```